mmj2jar/PATutorial/Page103.mmp

```diff
@@ -18,16 +18,23 @@ $( <MM> <PROOF_ASST> THEOREM=welcome  LOC_AFTER=ax-mp
 
  The Good News is that it is very safe. Unless you explicitly tell it
  to "Save", "Store" or "Load" using the File or TL menu items it updates
- nothing on your system. You cannot damage a Proof Worksheet by 
+ nothing on your system. You cannot damage a Proof Worksheet by
  experimenting with it -- unless you do File/Save...etc.
 
+ Also, if you really mess things up, you can use Edit/Undo
+ (control-Z). If you're running on Java 8, undo will undo a block of text,
+ while if you're running on Java 9 or later, undo will undo a
+ character at a time. You can use undo repeatedly to keep undoing more
+ until you get back to where you want to be. If you undo too much,
+ you can then use Edit/Redo (control-Y).
```

Comment:
I don't think the Java 9 behavior difference deserves a mention. It's undo either way, people will figure it out.

Reply:
I don't agree that people will figure it out. I thought undo was broken, and only realized what was going on when I read the source code for mmj2. I don't think most people will read source code.

Reply:
@david-a-wheeler Why did you think undo was broken? Did undo not undo under some circumstance?

Reply:
I pressed control-Z, didn't see a difference, and just assumed that undo had gotten broken.

Unfortunately I don't remember in exactly what circumstance that was, so I can't reproduce it. But I think part of the issue is that it's a lot harder to notice an undo if it does single characters.

So making it *clear* that undo is a single character (in at least some circumstances) is useful. Different behavior that depends on the Java version is unusual, and so while I think it's *okay*, I think it's useful to mention it to prevent confusion.